apps/documents/assets/DocumentManagement.jsx

```diff
@@ -0,0 +1,315 @@
+var api = require('adhocracy4').api
+var React = require('react')
+var django = require('django')
+var update = require('react-addons-update')
+var ChapterList = require('./ChapterList')
+var ChapterForm = require('./ChapterForm')
+var Alert = require('../../contrib/static/js/Alert')
+
+const DocumentManagement = React.createClass({
+  getInitialState: function () {
+    let chapters = this.props.chapters
+    if (!chapters || chapters.length === 0) {
+      chapters = [
+        this.getNewChapter(django.gettext('first chapter'))
+      ]
+    }
+
+    return {
+      chapters: chapters,
+      errors: null,
+      alert: null,
+      editChapterIndex: 0
+    }
+  },
+
+  maxLocalKey: 0,
+  getNextLocalKey: function () {
+    /** Get an artificial key for non-committed items.
+     *
+     *  The key is prefixed to prevent collisions with real database keys.
+     */
+    this.maxLocalKey++
+    return 'local_' + this.maxLocalKey
+  },
+
+  /*
+  |--------------------------------------------------------------------------
+  | Chapter state related handlers
+  |--------------------------------------------------------------------------
+  */
+
+  getNewChapter: function (name) {
+    return {
+      name: name,
+      key: this.getNextLocalKey(),
+      weight: 0,
+      paragraphs: []
+    }
+  },
+
+  handleChapterMoveUp: function (index) {
+    const value = this.state.chapters[index]
+    const diff = {$splice: [[index, 1], [index - 1, 0, value]]}
+    let editChapterIndex = this.state.editChapterIndex
```

Comment:
Tracking the current chapter by index seems tedious to me. Is it not possible to track it by something that does not change all the time (e.g. key/id)?